### DISCUSSION TITLE
Label for new option

### DISCUSSION THREAD

**Discussion by dragomano** [sa-si-dev/virtual-select#129] [Q&A] [answered]
https://codepen.io/dragomano/pen/oNpPvxw

Try to add this text - `fas fa-user fa-spin`
The option text displays correctly - with icon. But after adding this option, the icon (in the input field) is not displayed.
![](https://user-images.githubusercontent.com/229402/162999945-527646d3-f4db-4fac-bd19-3e3fee580045.png)

**Comment by sa-si-dev** [comment, marked as answer]
This is expected behaviour only. `labelRenderer` would be used only for rendering drop-down options, not for value text.
For new options, given text would be used as value text and label text.

**Reply by dragomano** [reply]
And how to display icons of new options in the input field?

**Reply by sa-si-dev** [reply]
I don't find any scenario where user would enter new option with icon related text.

If there is any scenario, please let me know, I will implement it as a new feature.

**Reply by dragomano** [reply]
I have the FontAwesome search input and I just want to see each object (added by user) with such structure:
```
{
      label: `<i class="${data.value}"></i> ${data.value}`,
      value: data.value,
      isNew: true
}
```

Is it possible?

**Reply by sa-si-dev** [reply]
> I have the FontAwesome search input and I just want to see each object (added by user) with such structure:
> ```
> {
>       label: `<i class="${data.value}"></i> ${data.value}`,
>       value: data.value,
>       isNew: true
> }
> ```
> 
> Is it possible?

As of now, label and value would be same for new options.

**Reply by sa-si-dev** [reply]
> I have the FontAwesome search input and I just want to see each object (added by user) with such structure:
> 
> ```
> {
>       label: `<i class="${data.value}"></i> ${data.value}`,
>       value: data.value,
>       isNew: true
> }
> ```
> 
> Is it possible?

You could update the structure of the selected values before sending them to your server in your code.
